packages/studioCMS/src/integrations/studioCMSDashboard/routes/dashboard/components/PagespeedInsights.astro

```diff
@@ -0,0 +1,160 @@
+        </div>
+        <div class="flex justify-center font-bold">Best Practices</div>
+    </div>
+
+    <div id="w-full flex flex-col">
+        <div class="flex justify-center">
+            <ProgressRing label="seo-score" id="seo-ring" value={0}>N/A</ProgressRing>
+        </div>
+        <div class="flex justify-center font-bold">SEO</div>
+    </div>
+
+</div>
+
+<Divider class:list="divider" />
+
+<div class="text-gray-500">
+    Test results are generated using Google's PageSpeed Insights API. If you would like to see more details for the Results above, please visit the <a class="text-blue-500" href={`https://pagespeed.web.dev/analysis?url=${encodeURIComponent(FetchURL.toString())}`} target="_blank" rel="noopener noreferrer">PageSpeed Insights Results</a>.
+</div>
+
+<VisuallyHidden>
+    <div id="lighthouse-fetch-url">{FetchURL}</div>
+</VisuallyHidden>
+
+<style scoped>
+    #perf-ring, #acc-ring, #bp-ring, #seo-ring {
+        --track-width: 1rem;
+        font-weight: bolder;
+        font-size: 2rem;
+    }
+</style>
+
+<script>
+import type { SlButton, SlProgressRing } from "astrolace:types";
+
+function generateLighthouseFetchUrl(url: string, strategy: 'mobile' | 'desktop' | 'mixed-content' = 'mobile'): string {
+    // Base URL for Google PageSpeed Insights API
+    const baseUrl: string = 'https://www.googleapis.com/pagespeedonline/v5/runPagespeed';
+    
+    // Final URL with encoded parameters
+    const fetchUrl: string = `${baseUrl}?url=${encodeURIComponent(url)}&fields=lighthouseResult%2Fcategories%2F*%2Fscore&prettyPrint=false&strategy=${strategy}&category=performance&category=pwa&category=best-practices&category=accessibility&category=seo`;
+    
+    return fetchUrl;
+}
+
+const urlToTest: string = document.getElementById('lighthouse-fetch-url')?.textContent || '';
+
+const lighthouseFetchUrl: string = generateLighthouseFetchUrl(urlToTest);
+
+const runTestButton = document.getElementsByName('run-test')[0] as SlButton;
+
+runTestButton.addEventListener('click', async (e) => {
+    e.preventDefault();
+
+    runTestButton.loading = true;
+    runTestButton.disabled = true;
+
+    console.log('PageSpeed Insights test started.', lighthouseFetchUrl)
+
+    const response = await fetch(lighthouseFetchUrl);
+
+    let performanceScore = 0;
+    let accessibilityScore = 0;
+    let bestPracticesScore = 0;
+    let seoScore = 0;
+
+    if (response.ok) {
+
+        const data = await response.json();
+
+        console.log('Successfully fetched data from the PageSpeed Insights API.', data);
+
+        performanceScore = data.lighthouseResult.categories.performance.score * 100;
+        accessibilityScore = data.lighthouseResult.categories.accessibility.score * 100;
+        bestPracticesScore = data.lighthouseResult.categories['best-practices'].score * 100;
+        seoScore = data.lighthouseResult.categories.seo.score * 100;
+
+        const perfRing = document.getElementById('perf-ring') as SlProgressRing
```

Comment:
`performanceRing` and come on no jokes here please, settle down everyone

Reply:
What is with your obsession of making LONG AS* NAMED STRINGS!? lol its just used to change the value of the ring on one page....

Reply:
[07f81d9](https://github.com/astrolicious/studiocms/pull/125/commits/07f81d9d66a3477a73ee1a4ce3e91ba85b26c905) enjoy the notes.... haha

Reply:
You laugh but that's literally 100x better